iceoryx_hoofs/CMakeLists.txt

```diff
@@ -82,6 +85,15 @@ iox_add_library(
         source/units/duration.cpp
 )
 
+configure_file("${CMAKE_CURRENT_SOURCE_DIR}/cmake/iceoryx_hoofs_deployment.hpp.in"
+  "${CMAKE_BINARY_DIR}/generated/iceoryx_hoofs/include/iceoryx_hoofs/iceoryx_hoofs_deployment.hpp" @ONLY)
+
+install(
```

Comment:
This is not required anymore. `iox_add_library` and `iox_add_executable` have an install directive which take care of the right prefix, project name etc.
I admit that they do not yet install generated hpp file (so it seems at least when I look at the code) but they seem to install all interface files which this file should be part of it.

So could you remove this install directive and check if the file is not already installed by `iox_add_library`. If not could you add an additional argument `INSTALL_ADDITIONAL_FILES` where this file is provided. Then you could add the correct prefix inside the `iox_add_library` thingy.
Whenever we decide to restructure this we only have to touch `iox_add_library` and nothing else but when there is an `install` directive outside it complicates things again.

Reply:
@elfenpiff I copied this from posh. I guess it's currently not possible. I'll have a look at it

Reply:
@elfenpiff unfortunately it is not easily doable in a clean way. I have an idea how to do it but I'm not sure if it works and it takes some time. Since we do it the same way in posh, would you be fine if I create an issue for this to solve it in both packages?

Reply:
I am fine with creating an issue. Lets merge this

Reply:
Done with #1687